### ISSUE TITLE
Puppet Master could not be reached.

### ISSUE THREAD

**Issue by filipmoons** [oscar-stack/oscar#31]
Dear Adrien,
Thanks for developing Oscar for us, I really would like to use it to experiment a bit. But, when I try to install it and run "vagrant up", whatever I try, I keep getting this message:

!! ERROR: Puppet Master at 'puppet:8140' could not be reached.
   Aborting installation as directed by answer file. Set
   'q_fail_on_unsuccessful_master_lookup' to 'n' if installation
   should continue despite communication failures.

I also tried to change the roles.yaml file to add the master to the (first and only) agent, but also that one failed.

Thanks in advance,
Filip Moons

**Comment by dec-pupop** [comment]
I am also having this issue, have you found solution for this?

**Comment by filipmoons** [comment]
Hello,
Actually I didn't find a solution for Oscar. I know that there is a way to get around this, because my mentor during my internship succeeded after trying a long time. But since my internship was almost over and I had not much time, I tried the Puppet-Sandbox of Aaron Bull Schaefer (https://github.com/elasticdog/puppet-sandbox) instead. This is basically a bunch of configuration files for vagrant to simulate an environment with a puppet master and one or more agents. It works smoothly and it suited my needs perfectly.

**Comment by dec-pupop** [comment]
Thanks for your update, I have raised it as an issue here also, hopefully I will get some response.
thanks for your advise on puppet sandbox, I tried it and it worked first time. 

I was hoping to use puppet enterprise which I'm not sure is possible with sandbox, need to do further investigation but this will work for now, thanks.

**Comment by weilera** [comment]
Don't know if it is still needed but anyway..
try to disable the firewall on the master node
`$ vagrant ssh master`
`sudo service iptables stop`
`exit`
Then provision first node
`$ vagrant provision first`

**Comment by TJM** [comment]
Right! The problem here is that the firewall is on by default, and the puppet enterprise installer is not managing it. No manual changes should be necessary, but for sure this one bit me too.

On CentOS 7, the workaround to disable the firewall is:
* `vagrant ssh master`
* `sudo systemctl disable firewalld`
*` sudo systemctl stop firewalld`
* `exit

Then, as noted you can `vagrant provision first` (or the name of your agent).
